### PR TITLE
fix oneshot params

### PR DESCRIPTION
**Pull request by bertyah**
I found that the current implementation did not work to modify the amount of items returned. I tried this fix and was able to get higher amounts back from the query.

### CODE REVIEW

**Comment by bertyah** [comment]
I also updated ResultReader to JSONResultReader because of the deprecation warning.

File: actions/search.py [hunk 1]
```diff
@@ -52,9 +52,11 @@ def run(self, instance, query, count):
                 "Failed to connect to Splunk Instance {} with error {}".format(splunk_config, err)
             )
 
-        result = self.service.jobs.oneshot(query, params={"output_mode": "json", "count": count})
-        reader = results.ResultsReader(result)
+
```
Comment (rush-skills) [inline actions/search.py]: Flake doesn't like the extra newline and fails the CI. Removing it should fix it.

> /home/runner/work/stackstorm-splunk/stackstorm-splunk/pack/actions/search.py:56:9: E303 too many blank lines (2)

File: actions/search.yaml [hunk 2]
```diff
@@ -20,3 +20,8 @@ parameters:
     required: true
     position: 2
     default: 100
+  offset:
+    type: integer
+    description: "A number that specifies the index of the first entity to return"
+    position: 3
+    default: 0
```
Comment (rush-skills) [inline actions/search.yaml]: Lets keep the newline

Comment (bertyah) [reply actions/search.yaml]: fixed!

File: actions/search.py (hunk 1 above)
Comment (bertyah) [reply actions/search.py]: fixed!

**Comment by bertyah** [comment]
CLA signed!

**Comment by bertyah** [comment]
We can close this PR because I submitted an updated one

**Pull request closed without merging**